src/ToeplitzMatrices.jl

```diff
@@ -43,7 +43,10 @@ end
 convert(::Type{Matrix}, A::AbstractToeplitz) = Matrix(A)
 
 # Fast application of a general Toeplitz matrix to a column vector via FFT
-function mul!(y::StridedVector{T}, A::AbstractToeplitz{T}, x::StridedVector, α::T, β::T) where T
+function mul!(y::StridedVector, A::AbstractToeplitz, x::StridedVector, α::Number, β::Number)
+    T = promote_type(eltype.([y, A, x, α, β])...)
+#   toreal = x -> (T <: Real) ? convert(T, x) : x # causes InexactError
+    toreal = x -> (T <: Real) ? real(x) : x
```

Comment:
What is the point of `toreal`?  Why make this an anonymous function? The previous version is simpler

Reply:
Ok, I will use the ternary instead.  Might be better for the compiler I guess, and branch prediction will be perfect.